data.sql

```diff
@@ -45,4 +45,69 @@ UPDATE animals SET owner_id = owners.id FROM owners WHERE name IN ('Charmander',
 
 UPDATE animals SET owner_id = owners.id FROM owners WHERE name IN ('Angemon','Boarmon') AND owners.full_name= 'Dean Winchester';
 
+COMMIT;
+
+--- DAY 4 ---
+
+/* Insert the following data for vets */
+INSERT INTO vets(name, age, date_of_graduation)
+VALUES 
+		('William Tatcher', 45, 'Apr 23, 2000'),
+		('Maisy Smith', 26, 'Jan 17, 2019'),
+		('Stephanie Mendez', 64, 'May 4, 1981'),
+		('Jack Harkness', 38, 'June 8, 2008');
+		
+/* Insert the following data for specialties */
+BEGIN;		
+	INSERT INTO specializations(vet_id, species_id)
+	SELECT vets.id, species.id FROM vets
+	JOIN species ON species.name = 'Pokemon' AND vets.name = 'William Tatcher';
+
+	INSERT INTO specializations(vet_id, species_id)
+	SELECT vets.id, species.id FROM species
+	JOIN vets ON vets.name = 'Stephanie Mendez';
+
+	INSERT INTO specializations(vet_id, species_id)
+	SELECT vets.id, species.id FROM vets
+	JOIN species ON species.name = 'Digimon' AND vets.name = 'Jack Harkness';
+COMMIT;
+
+/* function to save data into visits */
+
+CREATE OR REPLACE function save_visit_date(animal_name char(50), vet_name text ARRAY, vi_date date ARRAY) 
+		returns void
+		language plpgsql
+	as $$
+	declare 
+		vi_date_length numeric(10, 0) := 0;
+		vet_name_length numeric(10, 0) := 0;
+	begin
+		vi_date_length := array_length(vi_date, 1);
+		vet_name_length := array_length(vet_name, 1);
+				
+		if(vi_date_length > 0 AND vet_name_length > 0) then
+			for i in 1..vi_date_length loop
+				INSERT INTO visits(animals_id, vet_id, visit_date)
+				SELECT animals.id, vets.id, vi_date[i] FROM vets
+				JOIN animals ON animals.name = animal_name AND vets.name = vet_name[i];
+			end loop;
+		end if;
+	end;
+	$$
+
+/* Insert the following data for visits */
+BEGIN;
+	SELECT * FROM save_visit_date('Agumon', '{"William Tatcher", "Stephanie Mendez"}', '{"May 24, 2020", "Jul 22, 2020"}');
+	SELECT * FROM save_visit_date('Gabumon', '{"Jack Harkness"}', '{"Feb 2, 2021"}');
+	SELECT * FROM save_visit_date('Pikachu', '{"Maisy Smith", "Maisy Smith", "Maisy Smith"}', '{"Jan 5, 2020", "Mar 8, 2020", "May 14, 2020"}');
+	SELECT * FROM save_visit_date('Devimon', '{"Stephanie Mendez"}', '{"May 4, 2021"}');
+	SELECT * FROM save_visit_date('Charmander', '{"Jack Harkness"}', '{"Feb 24, 2021"}');
+	SELECT * FROM save_visit_date('Plantmon', '{"Maisy Smith", "William Tatcher", "Maisy Smith"}', '{"Dec 21, 2019", "Aug 10, 2020", "Apr 7, 2021"}');
+	SELECT * FROM save_visit_date('Squirtle', '{"Stephanie Mendez"}', '{"Sep 29, 2019"}');
+	SELECT * FROM save_visit_date('Angemon', '{"Jack Harkness", "Jack Harkness"}', '{"Oct 3, 2020", "Nov 4, 2020"}');
+	SELECT * FROM save_visit_date('Boarmon', '{"Maisy Smith", "Maisy Smith", "Maisy Smith", "Maisy Smith"}', 
+								  '{"Jan 24, 2019", "May 15, 2019", "Feb 27, 2020", "Aug 3, 2020"}');
+	SELECT * FROM save_visit_date('Blossom', '{"Stephanie Mendez", "William Tatcher"}', '{"May 24, 2020", "Jan 11, 2021"}');
```

Comment:
- [ ] According to [requirements](Insert the following data for visits:) you need to insert 20 visits, here you're inserting only 10. Please, check the requirements again and do the last visits to complete this project. 🌴

Reply:
Thanks, Victor for the review.

Let me explain why to answer is correct, the requirement is to insert 20 visits in the visit table so because there is so much data I find a way to do it in a much cleaner way with a SQL function I even make a reference to that function in the PR description https://github.com/miguelesco/vet-clinic/blob/f45653c4dda3ea08f477b4b9d5b80062fcfaba8f/data.sql#L77 

Well, that function takes 3 parameters the name of the animal as string, an ARRAY of the vet names, and another Array of the DATES so with that 3 parameters I can fill all the info thanks to a loop I made inside the function for more details anyone can review the logic of the function.

I invite you to test my code so you can be sure that the 20 rows are getting fulfilled with that 10 lines using the function.

You only run the create function first then the 10 lines of code that appeared in the comment.

For example, if you watch the line that starts for Pikachu you can see that right next to the coma there is the array of all the vets that Pikachu visits, and next to that the array of all the dates.

Reply:
hoppefully with this explanation I get approved

Reply:
I think the explanation is clear, thanks to share this with me. I appreciate it a lot. 😄

Reply:
> I think the explanation is clear, thanks to share this with me. I appreciate it a lot. 😄

No problem thanks for all